test/api/route_test.js

```diff
@@ -143,6 +156,40 @@ suite("api/route", function() {
       });
   });
 
+  test("query parameter", function() {
+    var url = 'http://localhost:23525/query-param/';
+    return request
+      .get(url)
+      .query({nextPage: '352'})
+      .end()
+      .then(function(res) {
+        assert(res.ok, "Request failed");
+        assert(res.text === "352", "Got wrong value");
+      });
+  });
+
+  test("query parameter (is optional)", function() {
+    var url = 'http://localhost:23525/query-param/';
+    return request
+      .get(url)
+      .end()
+      .then(function(res) {
+        assert(res.ok, "Request failed");
+        assert(res.text === "empty", "Got wrong value");
+      });
+  });
+
+  test("query parameter (validation works)", function() {
+    var url = 'http://localhost:23525/query-param/';
+    return request
+      .get(url)
+      .query({nextPage: 'abc'})
+      .end()
+      .then(function(res) {
+        assert(!res.ok, "Expected request failure!");
+        assert(res.status === 400, "Expected a 400 error");
+      });
+  });
```

Comment:
first test returns the querystring... but it's not very interactive... second test returns empty as no querystring is provided...